test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -317,137 +306,131 @@ TEST_P(OverrideHostTest, DrainingExcludedFromOverrideSet) {
 }
 
 TEST_P(OverrideHostTest, ClusterGoneHostStays) {
-  CreateAndStartBackends(2);
+  CreateAndStartBackends(3);
   const char* kNewCluster1Name = "new_cluster_1";
   const char* kNewEdsService1Name = "new_eds_service_name_1";
   const char* kNewCluster2Name = "new_cluster_2";
   const char* kNewEdsService2Name = "new_eds_service_name_2";
-  const uint32_t kWeight1 = std::numeric_limits<uint32_t>::max() / 2;
-  const uint32_t kWeight2 = std::numeric_limits<uint32_t>::max() - kWeight1;
+  const char* kNewCluster3Name = "new_cluster_3";
+  const char* kNewEdsService3Name = "new_eds_service_name_3";
+  const uint32_t kWeight1 = 3;
+  const uint32_t kWeight2 = 1;
   const double kErrorTolerance = 0.025;
-  const double kWeight2Percent =
-      static_cast<double>(kWeight2) / std::numeric_limits<uint32_t>::max();
+  const double kPercentage1 =
+      static_cast<double>(kWeight1) / (kWeight1 + kWeight2);
   const size_t kNumEchoRpcs =
-      ComputeIdealNumRpcs(kWeight2Percent, kErrorTolerance);
+      ComputeIdealNumRpcs(kPercentage1, kErrorTolerance);
   // Populate EDS and CDS resources.
-  SetEdsCdsResources(kNewCluster1Name, kNewEdsService1Name, 0, 1);
-  SetEdsCdsResources(kNewCluster2Name, kNewEdsService2Name, 1, 2);
+  SetCdsAndEdsResources(kNewCluster1Name, kNewEdsService1Name, 0, 1);
+  SetCdsAndEdsResources(kNewCluster2Name, kNewEdsService2Name, 1, 2);
   // Populating Route Configurations for LDS.
   SetListenerAndRouteConfiguration(
       balancer_.get(), BuildListenerWithStatefulSessionFilter(),
-      BuildRouteConfiguration(
+      BuildRouteConfigurationWithWeightedClusters(
           {{kNewCluster1Name, kWeight1}, {kNewCluster2Name, kWeight2}}));
   WaitForAllBackends(DEBUG_LOCATION, 0, 2);
-  auto session_cookie =
-      GetAffinityCookieHeaderForBackend(DEBUG_LOCATION, 1, 10);
-  ASSERT_FALSE(session_cookie.empty());
-  SetListenerAndRouteConfiguration(
-      balancer_.get(), BuildListenerWithStatefulSessionFilter(),
-      BuildRouteConfiguration({{kNewCluster1Name, kWeight1}}));
+  auto host2_in_cluster2_cookie =
+      GetAffinityCookieHeaderForBackend(DEBUG_LOCATION, 1);
+  ASSERT_FALSE(host2_in_cluster2_cookie.empty());
   // Cluster is gone
   balancer_->ads_service()->UnsetResource(kEdsTypeUrl, kNewEdsService1Name);
   balancer_->ads_service()->UnsetResource(kEdsTypeUrl, kNewEdsService2Name);
   // Both backends are in the same cluster
-  SetEdsCdsResources(kNewCluster1Name, kNewEdsService1Name, 0, 2);
-  WaitForAllBackends(DEBUG_LOCATION, 0, 2);
+  SetCdsAndEdsResources(kNewCluster1Name, kNewEdsService1Name, 0, 2);
+  SetCdsAndEdsResources(kNewCluster3Name, kNewEdsService3Name, 2, 3);
+  SetListenerAndRouteConfiguration(
+      balancer_.get(), BuildListenerWithStatefulSessionFilter(),
+      BuildRouteConfigurationWithWeightedClusters(
+          {{kNewCluster1Name, kWeight1}, {kNewCluster3Name, kWeight2}}));
+  WaitForAllBackends(DEBUG_LOCATION, 0, 3);
+  auto host3_in_cluster3_cookie =
```

Comment:
I don't think we need this check.  The goal here is to check what we do with the original cookie, not to get a new cookie.